timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -427,16 +417,26 @@ void __global__ k_nonbonded_unified(
                 coords,
                 params,
                 shared_box,
+                block_energy_buffer,
                 beta,
                 cutoff,
                 row_idxs,
                 ixn_tiles,
                 ixn_atoms,
                 du_dx,
-                du_dp,
-                u_buffer,
-                u_overflow_count);
+                du_dp);
         };
         tile_idx += stride;
     }
+    if (COMPUTE_U) {
+        // Sync to ensure the shared buffers are populated
+        __syncthreads();
+        if (threadIdx.x == 0) {
```

Comment:
I suspect this line might be the cause of the slowdown. Let's talk on Monday to see if we can speed this up.

Reply:
As of commit https://github.com/proteneer/timemachine/pull/1090/commits/db125b53b086bc8bf14ea337e997a391f8db1d69

```
dhfr-apo: N=23558 speed: 694.65ns/day dt: 2.5fs (ran 100000 steps in 31.10s)
dhfr-apo-barostat-interval-25: N=23558 speed: 622.25ns/day dt: 2.5fs (ran 100000 steps in 34.72s)
building a protein system with 1758 protein atoms and 7047 water atoms
hif2a-apo: N=8805 speed: 1179.01ns/day dt: 2.5fs (ran 100000 steps in 18.32s)
hif2a-apo-barostat-interval-25: N=8805 speed: 1002.58ns/day dt: 2.5fs (ran 100000 steps in 21.55s)
hif2a-rbfe: N=8840 speed: 892.35ns/day dt: 2.5fs (ran 100000 steps in 24.21s)
hif2a-rbfe-local: N=8840 speed: 1313.10ns/day dt: 2.5fs (ran 100000 steps in 16.45s)
solvent-apo: N=6282 speed: 1452.43ns/day dt: 2.5fs (ran 100000 steps in 14.87s)
solvent-apo-barostat-interval-25: N=6282 speed: 1261.00ns/day dt: 2.5fs (ran 100000 steps in 17.13s)
solvent-rbfe: N=6317 speed: 1175.46ns/day dt: 2.5fs (ran 100000 steps in 18.38s)
solvent-rbfe-local: N=6317 speed: 1408.38ns/day dt: 2.5fs (ran 100000 steps in 15.34s)
```